uiowa/uiowa
# Entity Usage Delete

Resolves https://github.com/uiowa/uiowa/issues/7768
Resolves https://github.com/uiowa/uiowa/issues/7644

Sorry, not sorry I've added a bit of scope-creep to this solution.

With 7768 we blindly accepted some entities to be tracked with entity_usage which is adding unnecessary rows to the table but is otherwise no visible to users. This also starts to cover fragments/layout builder which were previously unaccounted for.

Updated node_revision_delete because we had to upgrade to their rewrite for d10 support? in which some of the previous functionality was lost. Moving from an alpha to an RC brings back the "delete prior revisions" functionality which makes it easier for folks to clean up revisions before trying to delete media.





# How to test

- On main, `ddev blt di --site=default` and try to delete media associated with layout builder. Ya can't. Try to do the same for a v2 site in which the media is nested within some paragraphs. Ya can't.
- On `entity_usage_delete` try the delete operations again after doing the normal expected workflow. https://sitenow.uiowa.edu/documentation/media#delete-media, you should be able to. Also note the ease in which you can delete prior revisions all in one go versus having to delete each revision one by one.
- Go create a prefooter textarea with a media item added to it. On the media item you should see that there is entity usage coverage there now for the block_content, but that you can't delete the media until it is removed from the fragment. Fragments have revisions ui but only one revision is tracked in the ui.
- For each media that you are able to delete. The corresponding file_managed row should get a status of 0 which means it will be cleaned up in about six hours on cron. file_usage should also clear of the `fid` as media is no longer tied to it. You can expedite the process by changing the modified timestamp and running cron manually.
- To understand better the changes to config, I found it helpful to look at the entity_usage table on the default site. There you will see what it means by source/target entities.
